### ISSUE TITLE
Widgets don't work with `display(...)`, and instead only appear as `repr(...)` text

### ISSUE THREAD

**Issue by posita** [jupyterlite/jupyterlite#1458]
I can't figure out how to build a Pyodide-based site that renders widgets. It renders the `repr(...)` of the widget instead:

![Screenshot from 2024-08-12 20-13-41](https://github.com/user-attachments/assets/d7c2561b-f58a-481b-8244-5195748d7376)

Current attempt: https://posita.github.io/dyce-notebooks/lab/
Source: https://github.com/posita/dyce-notebooks

To repro:

1. Go to https://posita.github.io/dyce-notebooks/lab/ and create a new notebook.
2. Copy the following into the first cell:
``` py
# Install additional requirements if necessary
import warnings
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    try:
        import ipywidgets
    except (ImportError, ModuleNotFoundError):
        # See <https://jupyterlite.readthedocs.io/en/stable/howto/configure/simple_extensions.html#avoid-the-drift-of-versions-between-the-frontend-extension-and-the-python-package>
        requirements = ["ipycanvas==0.13.2", "ipyevents==2.0.1", "ipympl==0.9.4", "ipywidgets==8.1.3"]
        try:
            import piplite ; await piplite.install(requirements, keep_going=True)
            # Work around <https://github.com/jupyterlite/jupyterlite/issues/838>
            import matplotlib.pyplot ; matplotlib.pyplot.clf()
        except ImportError:
            import pip ; pip.main(["install"] + requirements)
import ipywidgets
display(ipywidgets.IntSlider())
```
3. Run the cell. You won't see a slider. You'll see the text output `IntSlider(value=0)`. If I run the same code in non-Lite Jupyter Lab, it works.

**Comment by posita** [comment]
It looks like I needed to install `jupyterlab_server` instead of `jupyter_server`. Is this documented somewhere prominent?

**Comment by posita** [comment]
Looks like I needed `jupyterlab_server` (but only had `jupyter_server`) as a build dependency. Is that documented somewhere prominently?

**Comment by jtpio** [comment]
> Looks like I [needed `jupyterlab_server` (but only had `jupyter_server`) as a build dependency](https://github.com/posita/dyce-notebooks/commit/883d3c50db3a2b59dc07fc16e676b26c5af61a45#diff-fefb2e8eecd362019729766b0bea79e46ed583b2e5b0af47dde7700c7c021a02). Is that documented somewhere prominently?

Right, there is a bit of documentation about this in https://jupyterlite.readthedocs.io/en/stable/reference/cli.html#addon-dependencies. But it could be made more visible.

There is also ongoing work to improve the error message when that dependency is not installed: https://github.com/jupyterlite/jupyterlite/pull/1401

**Comment by posita** [comment]
> **EDIT**: Ah, re-reading the message and the looking at the diff, it looks like the `ipywidgets` dependency was missing, which makes more sense because `jupyterlab_server` should not have any effect with widgets.

Thanks! Is https://github.com/jupyterlite/jupyterlite/issues/1453#issuecomment-2271657763 related then? That was from another issue where I was missing a dependency (that `jupyterlab_server` would have indirectly satisfied), but there wasn't really any indication or warning.
